ibridges/data_operations.py

```diff
@@ -245,6 +247,12 @@ def _upload_collection(session: Session, local_path: Union[str, Path],
             _obj_put(session, source, dest, overwrite, resc_name, options)
         except irods.exception.OVERWRITE_WITHOUT_FORCE_FLAG:
             warnings.warn(f'Upload: Object already exists\n\tSkipping {source}')
+        except KeyError as e:
+            if ignore_err is True:
+                warnings.warn(f'Upload failed: {source}\n'+repr(e))
+            else:
+                print(f'Upload failed: {source}')
```

Comment:
I think we should give the path of the failure within the error itself. Otherwise it will most likely be missed by the user. Additionally, I don't think a KeyError makes any sense here. Perhaps it's a RunTimeError? So I think we should reraise instead of using the original error.

Reply:
No unfortunately not. The irods server encodes all failures as negative integer numbers, which one can look up with the icommands `ierror <number>`. The python-irodsclient translates some of them to own errors like CAT_INVALID_USER. All those erros that are not translated will result in a dictionary key error :(
https://github.com/irods/python-irodsclient/blob/ed2e73cb79dcf08efedeed4a19f3e2a43db90f9c/irods/exception.py#L170C5-L170C49

Otherwise, I agree with you and maybe we need to creaye an IBRIDGES_SERVER_ERROR message for those errors.